Refactor Settings UI

Vendicated opened the pull request.

Vendicated commented:
I HATE CIRCULAR DEPENDENCIES 
![image](https://github.com/user-attachments/assets/10ccbe13-0b55-4ff9-82f5-cc147c776299)

Vendicated commented on src/api/Badges.ts:
```diff
@@ -16,11 +16,13 @@
 */
 
+// DO NOT REMOVE UNLESS YOU WISH TO FACE THE WRATH OF THE CIRCULAR DEPENDENCY DEMON!!!!!!!
+import "~plugins";
```
like why in the world does this have to be precisely here lmao

i tried adding it at the top of the api index.ts file but no dice

Vendicated commented:
I'm considering changing how settings work. I want to remove the Save button and instead make all changes immediately apply, to reduce complexity and because personally I don't think it's very useful to have

khcrysalis commented:
Wouldn't that require a restart for every single toggle though?

Vendicated commented:
no

sadan4 commented:
> I'm considering changing how settings work. I want to remove the Save button and instead make all changes immediately apply, to reduce complexity and because personally I don't think it's very useful to have

I like this idea. More often than I'd like to admit, I click out instead of clicking the save button, then wonder why nothing is working.

Vendicated commented:
I refactored the Settings and removed the saving process

Would be useful if people could test it :+1:

also seems like shiki settings are broken with this

Vendicated commented:
Okay, should be ready I think!

Vendicated commented:
I need people to test this change and just play around with settings & plugin settings a bunch and make sure everything still works fine
